content/en/agent/proxy.md

```diff
@@ -142,17 +142,25 @@ Do not forget to [restart the Agent][1] for the new settings to take effect.
 
 ## HAProxy
 
-[HAProxy][1] is a free, fast, and reliable solution offering proxying for TCP and HTTP applications. While HAProxy is usually used as a load balancer to distribute incoming requests to pools servers, you can also use it to proxy Agent traffic to Datadog from hosts that have no outside connectivity.
-
-This is the best option if you do not have a web proxy readily available in your network, and you wish to proxy a large number of Agents. In some cases, a single HAProxy instance is sufficient to handle local Agent traffic in your network-each proxy can accommodate upwards of 1000 Agents. **Note**: This figure is a conservative estimate based on the performance of m3.xl instances specifically. Numerous network-related variables can influence load on proxies. As always, deploy under a watchful eye. See the [HAProxy documentation][2] for additional information.
+[HAProxy][1] is a free, fast, and reliable solution offering proxying for TCP and HTTP applications. While HAProxy is usually used as a load balancer to distribute incoming requests to pools servers, you can also use it to proxy Agent traffic to Datadog from hosts that have no outside connectivity :
 
 `agent -> haproxy -> Datadog`
 
+This is the best option if you do not have a web proxy readily available in your network, and you wish to proxy a large number of Agents. In some cases, a single HAProxy instance is sufficient to handle local Agent traffic in your network as each proxy can accommodate upwards connections of 1000 Agents. **Note**: This figure is a conservative estimate based on the performance of m3.xl instances specifically. Numerous network-related variables can influence load on proxies. As always, deploy under a watchful eye. See the [HAProxy documentation][2] for additional information.
+
+The communication between HAProxy and Datadog will always be encrypted and it is recommended to encrypt the communication between the Agent host and the HAProxy host if they are not part of the same local network.
+In order to encrypt data between the Agent and HAProxy, you will need to create a Subject Alternative Name (SAN) certificate for the HAProxy host. This certificate should contain both the public certificate and private key. See this [HAProxy blog post][3] for more information.
+
 ### Proxy forwarding with HAProxy
 
 #### HAProxy configuration
 
-HAProxy should be installed on a host that has connectivity to Datadog. Use the following configuration file if you do not already have it configured.
+HAProxy should be installed on a host that has connectivity to Datadog. Use one of the following configuration files if you do not already have it configured.
+
+**Note**: It is recommended to use the `HTTPS` configuration file if the Agent and HAProxy are not part of the same local network.
```

Comment:
We should add a note here and in https version for users to replace all `<VARIABLE_NAME>` sections to prevent them from blindly copying them. This should probably be reflected in NGINX doc wording too.